chia/full_node/block_store.py

```diff
@@ -60,11 +60,14 @@ async def create(cls, db_wrapper: DBWrapper):
             )
 
             # Height index so we can look up in order of height for sync purposes
-            await self.db.execute("CREATE INDEX IF NOT EXISTS full_block_height on full_blocks(height)")
+            await self.db.execute("CREATE INDEX IF NOT EXISTS height on full_blocks(height)")
             await self.db.execute(
-                "CREATE INDEX IF NOT EXISTS is_fully_compactified on full_blocks(is_fully_compactified)"
+                "CREATE INDEX IF NOT EXISTS is_fully_compactified ON"
+                " full_blocks(is_fully_compactified, in_main_chain) WHERE in_main_chain=1"
```

Comment:
This index only use is for get_random_not_compactified? Adding height should improve lookup time for this query as all data is then already part of the index.

Reply:
right. On the other hand, it would make the index larger, but maybe not very much larger. it's worth an experiment

Reply:
It would really pay back, when you are able to only use one index [like mentioned here](https://github.com/Chia-Network/chia-blockchain/pull/9550#discussion_r768522270).

It did some runs of your benchmarks on remove-block-records-table branch (block_store.py):
v1 with current index: ~37s
v2 with current index: ~48s
v1 with index on full_blocks(height, is_fully_compactified): ~17s
v2 with index on full_blocks(height, is_fully_compactified): ~13s

Didn't use in_main_chain yet.